File: app/src/main/java/com/example/petridelivery/lists/adapters/abs/DiffItemCallback.java
package com.example.petridelivery.lists.adapters.abs;

import android.annotation.SuppressLint;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public abstract class DiffItemCallback<T> extends DiffUtil.ItemCallback<T> {
    @SuppressLint("DiffUtilEquals")
    @Override
    public boolean areContentsTheSame(@NonNull T oldItem, @NonNull T newItem) {
        return oldItem.equals(newItem);
    }
}

File: app/src/main/java/com/example/petridelivery/lists/adapters/abs/DiffItemCallback.kt
package com.example.petridelivery.lists.adapters.abs

import android.annotation.SuppressLint
import androidx.recyclerview.widget.DiffUtil

abstract class DiffItemCallback<T> : DiffUtil.ItemCallback<T?>() {
    @SuppressLint("DiffUtilEquals")
    override fun areContentsTheSame(oldItem: T, newItem: T): Boolean {
        return oldItem == newItem
    }
}